astropy/modeling/core.py

```diff
@@ -2462,8 +2460,16 @@ def render_model(model, arr=None, coords=None):
             assert coords[0].shape == arr.shape
         else:
             arr = np.zeros(coords[0].shape)
+    
+    if arr is not None:
+        arr = arr.copy()
+        # Check dimensions match model
+        assert arr.ndim == model.n_inputs
 
     if bbox is not None:
+        if wcs is not None:
+            bbox = np.array(wcs.wcs_world2pix(*[bbox[1],bbox[0],0]))[::-1]
+
```

Comment:
I need to add a check here that gets the limits right if the box overlaps the 0/360 RA or -90/90 Dec boundaries.